pkg/sdkserver/sdkserver.go

```diff
@@ -762,20 +784,196 @@ func (s *SDKServer) GetCounter(ctx context.Context, in *alpha.GetCounterRequest)
 	if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
 		return nil, errors.Errorf("%s not enabled", runtime.FeatureCountsAndLists)
 	}
-	// TODO(#2716): Implement me
-	return nil, errors.Errorf("Unimplemented -- GetCounter coming soon")
+
+	s.logger.WithField("name", in.Name).Info("Getting Counter")
+	s.gsUpdateMutex.RLock()
+	defer s.gsUpdateMutex.RUnlock()
+
+	// If we have made updates then return the SDK view of the Counter, otherwise get the counter from the gameserver
+	if counterUpdated, ok := s.gsCounterUpdates[in.Name]; ok {
+		s.logger.WithField("Get Counter from SDK Server", counterUpdated).Debugf("Got Counter %s", in.Name)
+		return &alpha.Counter{Name: in.Name, Count: counterUpdated.count, Capacity: counterUpdated.capacity}, nil
```

Comment:
I'm not sure how this applies to the gRPC standard, but at this point,  I feel like we should (assuming there is a diff) get the latest gameserver and apply any diff that exists/set overlap over the top of the existing counts, or possible values that are still in the batch pipeline.

The other thing we can do (with the suggestions above), is check the `Generation` stored with each processed batch request, and if `Generation` is greater than the one from `s.gameserver().ObjectMeta.Generation` we know we'll want to apply it to `gs.gameServer()`, since our local cache has yet to be populated with our batched changes.

(I hope that makes sense).

Reply:
And what if the diff can't be applied to the newest version of the gameserver? It implies we'd have to do updateCounter validation on every Get request.

Reply:
We'll have to update this code regardless given that the counterUpdateRequest will have possible nil values for count and capacity (since with the code changes those fields will represent countSet and capacitySet only).

I'm leaning towards not grabbing changes from the SDK server at all and just getting the GameServer, otherwise we'd need to do update validation in the GetCounter request.